src/components/AllProductsListSection/AllProductsListSection.tsx

```diff
@@ -68,6 +77,9 @@ export default function AllProductsListSection() {
                     favourite={product.is_favorite}
                     firstProduct={index === 0}
                     lastProduct={index === array.length - 1}
+                    id={product.id}
+                    cart={cart}
+                    setCart={setCart}
                   />
```

Comment:
given that this component has many props that are attributes of product I'd suggest having a product prop and destructure it inside the component

also a nit pick but it's a better practice to name props that trigger actions up the component tree like `onChangeCart` instead of setCart. Then is good to separate the callback used to set the state and the callback for the prop, that means either define a `handleCartChange` function in the ProductList component and set the state there or define a function right where the callback for the prop is defined with the new value 
This makes it easier to read and maintain 

```suggestion
                 <GeneralProductCard
                    firstProduct={index === 0}
                    lastProduct={index === array.length - 1}
                    product={product}
                    onChangeCart={() => setCart(someNewCartValue)}
                  />
```